### DISCUSSION TITLE
v3: Line breaks in the middle of a markdown link cause broken URLs

### DISCUSSION THREAD

**Discussion by rcdailey** [facebook/docusaurus#10067] [Q&A]
I'm upgrading to v3, and `yarn build` reports errors about broken links. In my case, I had the following:

```md
[application data
directory](/file-structure.md#appdata-directory)
```

I've trimmed this example, but for line-length reasons, this particular link was wrapped inside the square brackets. In Docusaurus v2, this caused no issue.

In v3, I get a report about a broken link:

```
  - Broken link on source page path = /wiki/cli/common/:
     -> linking to /file-structure.md/#appdata-directory
```

In preview mode with `yarn start`, I see this *broken* URL when I hover over the link:

```
http://localhost:3000/file-structure.md/#appdata-directory
```

If I remove the line break to make the markdown look like this:

```md
[application data directory](/file-structure.md#appdata-directory)
```

I get a correct URL:

```
http://localhost:3000/wiki/file-structure/#appdata-directory
```

I'm not sure why it's stripping off the `/wiki` portion; which I have set in my config:

```
  presets: [
    [
      'classic',
      /** @type {import('@docusaurus/preset-classic').Options} */
      ({
        docs: {
          routeBasePath: '/wiki',
  ...
```

**Comment by slorber** [comment]
Yes this is a known issue already reported in https://github.com/facebook/docusaurus/issues/9523

This should be fixed by using a real Markdown parser to process such links (currently we use line)-by-line regexp)
Track resolution here: https://github.com/facebook/docusaurus/issues/9048
